libvips/foreign/webp2vips.c

```diff
@@ -470,21 +470,32 @@ read_header( Read *read, VipsImage *out )
 		vips_image_set_int( out, 
 			VIPS_META_PAGE_HEIGHT, read->frame_height );
 
-		/* We must get the first frame to get the delay. Frames number
-		 * from 1.
-		 */
-		if( WebPDemuxGetFrame( read->demux, 1, &iter ) ) {
-			read->delay = iter.duration;
+		if ( read->frame_count > 1) {
+			read->delays = (int *) g_malloc( read->frame_count * sizeof(int) );
+
+			for( int i = 0; i < read->frame_count; i++ ) {
+				if( WebPDemuxGetFrame( read->demux, i + 1, &iter ) ) {
+					read->delays[i] = iter.duration;
+				} else {
+					read->delays[i] = 0;
+				}
+			}
 
 #ifdef DEBUG
-			printf( "webp2vips: duration = %d\n", read->delay );
+		for( int i = 0; i < read->frame_count; i++ ) {
+			printf( "webp2vips: frame = %d; duration = %d\n", i + 1, read->delays[i] );
+		}
 #endif /*DEBUG*/
 
+			vips_image_set_array_int( out, "delay", read->delays, read->frame_count );
+			g_free( read->delays );
```

Comment:
This will produce use-after-free errors. This should probably be a `VIPS_FREE()` in `read_free()`.

Reply:
Use after free when you access the array on the next line, I mean.